src/main/scala/org/citegraph/analytics/Analytics.scala

```diff
@@ -0,0 +1,33 @@
+package org.citegraph.analytics
+
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+class Analytics(sparkSession: SparkSession, citationsDF: DataFrame, publishedDatesDF: DataFrame) {
+
+  /**
+   * Finds the density of both nodes n(t) and edges e(t) for each year t.
+   */
+  def findDensitiesByYear(): Unit = {
+
+    /*
+    Find node density by year - group by year and count records:
+      +----+-----+
+      |year|count|
+      +----+-----+
+      |1997| 4252|
+      |1994| 2894|
+      |1996| 3939|
+      |... |...  |
+      +----+-----+
+     */
+    publishedDatesDF.groupBy(col("year"))
+      .count()
+      .show()
```

Comment:
I think we can probably get this to print in the same format as the template in the assignment description, but this is good for now.

Reply:
Noted; I've added a .csv saver called `DataFrameSaver` under pkg `saving`. It coalesces to 1 worker, can sort on a column (in this case, "year"), and saves to HDFS output dir you specify in program args.

That 'findDensitiesByYear()` function is not complete; we'll need to add the other two columns and name them appropriately